httpe/httpe.go

```diff
@@ -1,14 +1,35 @@
-// Package httpe provides http server utilities.
+// Package httpe provides a model of HTTP handler that returns errors.
+//
+// net/http provides a Handler interface that requires the handler to write
+// errors to the provided ResponseWriter. This is different to the usual Go way
+// of handling errors that has functions returning errors, and it makes normal
+// http.Handlers a bit cumbersome.
+//
+// This package provides a HandlerE interface with a ServeHTTPe method that has
+// the same signature as http.Handler.ServeHTTP except it also returns an
+// error. A separate error handler can be bound to the HandlerE using
+// NewHandler() and turn the HandlerE into an http.Handler.
+//
+// As well as making handler code a little simpler, separating the ErrWriter
+// allows for common error handling amongst disparate handlers.
+//
+// The default ErrWriter writes the error to the ResponseWriter with a 500
+// status code for unknown errors, or with the status code encoded in the error
+// for StatusErrors. Additionally, if the error is not a client error, no
```

Comment:
Consider adding "wrapping a StatusError" 
e.g. in your client code use 
    return fmt.Errorf("%w: not a number", httpe.ErrBadRequest)

Reply:
I have re-written this paragraph completely. I do not use code with `%w` as you suggested, but just use the term "wrap". Let me know what you think of this now.